### PR TITLE
Unit test fixes and run them as a GH action

### PR DESCRIPTION
**Pull request by cameronr**
I somehow accidentally removed the whitespace for the Makefile step that installs plenary. It should auto install now. Let me know if it doesn't work for you!

Also, what do you think about running the tests on commit? I'm not sure how GH does it but I assume something like that would be possible?

### CODE REVIEW

**Comment by cameronr** [comment]
This ended up being a bit more involved than I thought it would. Fixing the referenced bugs was easy but I noticed places where the path separators were being inconsistently added which made it hard to follow why things were happening and would cause the alternate session to be effectively the same as the current one.

**Comment by cameronr** [comment]
I'm working on the GH action to automatically run the unit tests. I'll add that once I have it working

**Comment by rmagatti** [comment]
Yeah a GH action to run the tests on push to PRs would be ideal 👍 great work @cameronr 💪

**Comment by cameronr** [comment]
I think this group of changes is all set. It'll run the tests on push and on PRs and test against 0.7.2, 0.8.3, 0.9.5, 0.10.0, and nightly.

**Review by rmagatti — APPROVED** [review]
